### ISSUE TITLE
Web - Unable to type when popover is open from emoji reaction

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#27250]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open the website
2. Open any chat from LHN
3. Send a message (it can be a text)
4. React an emoji for a message
5. Open popover from emoji reaction by clicking right-click
6. Start typing
7. Observe the result

## Expected Result:
 It should be able to type in the composer when a popover is open from an emoji reaction.

## Actual Result:
Unable to type in the composer when the popover is open from emoji reaction

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.68.12
**Reproducible in staging?:** y
**Reproducible in production?:** n
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/1d297ead-9607-4bcb-9564-a385a3edb2e0

https://github.com/Expensify/App/assets/93399543/85592ba7-a3f0-4a18-a9ba-5f51649dd3de

**Expensify/Expensify Issue URL:**
**Issue reported by:** @misgana96
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1694508014533939

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by jscardona12** [comment]
Can't reproduce on dev

**Comment by paultsimura** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.


### What is the root cause of that problem?
The text input blurs when the emoji bubble is right-clicked:

https://github.com/Expensify/App/blob/c17a085dac644ac761119388b04daad0382b6d5d/src/components/Reactions/EmojiReactionBubble.js#L75-L83

### What changes do you think we should make in order to solve the problem?
Remove the right-click check in the `onMouseDown`
```diff
            onMouseDown={(e) => {
-               // Allow text input blur when emoji reaction is right clicked
-               if (e && e.button === 2) {
-                  return;
-               }

                // Prevent text input blur when emoji reaction is left clicked
                e.preventDefault();
            }}
```



### What alternative solutions did you explore? (Optional)

**Comment by dangrous** [comment]
Yeah with that comment there it really feels like expected behavior - I'm syncing with the team to confirm but I think we'll be able to close this out.

**Comment by misgana96** [comment]
I think typing in the composer when popover is open from emoji reaction is expected behaviour

**Comment by dangrous** [comment]
Due to https://github.com/Expensify/App/pull/26826

**Comment by b4s36t4** [comment]
I'm not able re-produce this popover, how to enable this type of popup for reactions?

I think when a popover is opened user should not be able to type.

**Comment by b4s36t4** [comment]
I have disabled it because, I was able to type in composer even when Attachment popup is opened, call option menu is also opened.

**Comment by dangrous** [comment]
discussing in Slack [here](https://expensify.slack.com/archives/C01SKUP7QR0/p1694539186210669?thread_ts=1694529057.820779&cid=C01SKUP7QR0) to figure out expected behavior

**Comment by b4s36t4** [comment]
Also @aimane-chnaif I don't think it possible to have focus when a popover is opened. We remove focus from composer when a popover/modal so I think the expected behaviour is user shouldn't have any power to type when any modal is opened.

**Comment by b4s36t4** [comment]
I think this is the only place we keep composer focused, in all other places like even for emoji popup we remove focus.

**Comment by b4s36t4** [comment]
in my view weather if we have any popup opened we should never focus the composer.

**Comment by b4s36t4** [comment]
it doesn't make any sense to allow few popups (this GH & and other issue) & doesn't allow few popups (call popover, attachment popover).

**Comment by b4s36t4** [comment]
<img width="261" alt="Screenshot 2023-09-12 at 11 21 43 PM" src="https://github.com/Expensify/App/assets/59088937/cc42730a-8980-430e-b98a-e22387e07221">

<img width="516" alt="Screenshot 2023-09-12 at 11 21 39 PM" src="https://github.com/Expensify/App/assets/59088937/70ade4f4-490d-496e-a655-e7a45425a001">

I don't think these two are modals.

**Comment by b4s36t4** [comment]
The Root cause of that issue is also `non-modals`. I thought my PR would raise a regression so I updated that condition as well.

> Those are non-modals of course. When you type something, you will notice that composer is focused in production app.

This is bug right?

**Comment by b4s36t4** [comment]
> This is bug right?

if that is expected, I touched it unknowingly caused the regression.

**Comment by dangrous** [comment]
Okay so we're gonna remove the deployblocker label here, and keep this open until we sort out which direction we want to move in. I'm of the mind to disallow typing on any pop up, but we want to hear some more thoughts from the team! Stay tuned.

**Comment by paultsimura** [comment]
Just to sort some things out: if it appears that the issue is still relevant – will the proposals here potentially be reviewed, or it will it move to @b4s36t4?
So that I could know whether to unsubscribe from this issue's updates, thanks.

**Comment by b4s36t4** [comment]
99% I'm gonna tackle this issue (this will get closed) if we choose to keep the few popup open and allow typing. If we're going to disallow all popups to not focus there is nothing we're gonna do.
